### PR TITLE
Expand security privacy considerations, emphasize minimum viable features

### PR DESCRIPTION
**Pull request by tantek**
I've tried to expand security and privacy considerations according to the question I initially posed about doing what we can to reduce overall security surface, of WebAPIs, and the web platform in general. Have posted a blog post on this as well at http://tantek.com/2015/068/b1/security-towards-minimum-viable-web-platform

### CODE REVIEW

File: security-privacy/index.src.html [hunk 1]
```diff
@@ -317,10 +317,59 @@ <h3 id="considerations">
...
+  
   Documenting the various concerns that have cropped up in one form or another
   is a good way to help implementers and authors understand the risks that a
   feature presents, and ensure that adequate mitigations are in place.
+  
+  If it seems like a feature does not have security or privacy impacts, 
+  then say so inline in the spec section for that feature:
+  
+  <blockquote>
+  There are no known security or privacy impacts of this feature.
```
Comment (marcoscaceres) [inline security-privacy/index.src.html]: Scary. I've never encountered such a feature.

File: security-privacy/index.src.html [hunk 2]
```diff
@@ -317,10 +317,59 @@ <h3 id="considerations">
...
+  even if only for the reasons of reducing and minimizing 
+  security/privacy attack surface(s).
+  
+  By doing so we can reduce the overall security (and privacy) attack surface 
+  of not only a particular feature, but of a module (related set of features), 
+  a specification, and the overall web platform. 
+  Ideally this is one of many motivations to reduce each of those to the minimum viable:
+  <ol>
+    <li>
+    Minimum viable feature:
```
Comment (marcoscaceres) [inline security-privacy/index.src.html]: Maybe convert this to a definition list (`dl`) with corresponding  `dt`s and `dd`s?

**Comment by marcoscaceres** [comment]
I'm sympathetic to specifying minimal viable features (not only for privacy and security reasons, but also because it aligns firmly with the principles of the extensible web manifesto). However, the above should provide more guidance on how to continue to evolve features of the platform. 

It's pretty well understood, IMO, by the community that adding anything to the platform adds exponential complexity to maintenance, security, privacy, interoperability, etc. as those features interact with the rest of the platform.

**Comment by tantek** [comment]
@marcoscaceres IMO "how to continue to evolve features of the platform" is out of scope for a document on security. I mention minimizing because that directly impacts reducing the security attack surface.

And I have to completely disagree with your second paragraph "It's pretty well understood, IMO, by the community... " - IMO the opposite is true. The community as a whole developing Web APIs (perhaps even the open web platform as a whole) is more often than not *overdoing* it with needless complexity and featuritis, apparently sacrificing maintenance, security, privacy, interoperability. This is why I am ringing the alarm bell on this one.

**Comment by mikewest** [comment]
(Sorry! I remember we emailed about this, and then I got distracted, and never came back to this PR. My apologies; I appreciate the work you've done, and will try to be more responsive going forward.)

I think I agree with the core sentiment you're expressing in this PR: guilty until proven innocent is a reasonable approach to use when considering new platform features. I'm not sure I agree that this document is the right place to espouse the viewpoint that the platform should err towards minimum viable, but until there's a better document for that sentiment, expressing it here in some form is pretty reasonable.

**Comment by tantek** [comment]
Thanks @mikewest! I appreciate the "until there's a better document" methodology and at least capturing the thinking in some form. I didn't bother with updating the (presumably Bikeshed?) generated index.html, assuming you might be merging other patches and it would be better to leave updating of that to you.

**Pull request merged**